src/templates/head.html

```diff
@@ -20,6 +20,13 @@
   <link rel='apple-touch-icon' href='app/img/apple-touch-icon.png' />
   <link rel='apple-touch-icon-precomposed' href='app/img/apple-touch-icon-precomposed.png'>
   <title>{{ block('title') }} - eLabFTW</title>
+
+  {% if App.Users.userData.use_ove %}
+      <!-- load open vector editor -->
+      <link rel='stylesheet' type='text/css' href='app/ove/main.css?v={{ v }}'>
```

Comment:
the css should be included from main.scss
and ideally the js is in the bundle.

but damn that's a big file. 3.1 Mo for the minified version!!!!! WTF. The source file is 214.000+ lines! It's crazy!

But looking into the file, it looks like it's including everything. Like lodash and a bunch of other utilities. This is not the way to go. There is a reason we use webpack.

So now I'm wondering: maybe building this umd file is a mistake. What we want is an exported module with only the minimal code for displaying a plasmid. Maybe edit it too. Looking at the source code of the editor, the code is modern with es6 modules, so it should be possible to only import what we want and use the minimum. But then it pulls all react stuff and shit. Damn it's a hard problem....

Reply:
My rational and priority was to keep out the flood of dependencies and avoid the complexity of setting it up correctly with webpack and es6 modules based on [your comment](https://github.com/elabftw/elabftw/issues/277#issuecomment-691740704).
However, that comes at the price of having all the stuff like lodash included in the umd version and load it twice if it is also included in the elabftw webpack build.
But is that really a problem?

The way I implemented it keeps it very modular which also makes it easy to switch off the plasmid viewer and avoid loading the 3.1 Mb (I didn't know that it is mégaoctet in french) if it is switched off.

On the other side you are right about that the code is modern with es6 modules and it seems feasible to customize it to the specific needs (e.g. only build a viewer).

I guess you need to make a call and decide which way you wanna go.

Reply:
My goal here is to aim for the best implementation possible. I understand that the UMD version has to contain half of npmjs.com to work.

I'll need more time to think about it. I also want to try other approaches, like maybe having a lightweight viewer (using some code from OVE), without all the react stuff.

IMHO, only the viewer is really needed. I think an editor is overkill, as an external app should be used for that. So I'm wondering if it is feasible to extract the viewer part and only include minimal custom code in eLab to view plasmid files.